internal/zinc-core/src/main/scala/sbt/internal/inc/AnalysisStore.scala

```diff
@@ -9,30 +9,34 @@ package sbt
 package internal
 package inc
 
-import xsbti.compile.{ CompileAnalysis, MiniSetup }
+import java.util.Optional
 
-trait AnalysisStore {
-  def set(analysis: CompileAnalysis, setup: MiniSetup): Unit
-  def get(): Option[(CompileAnalysis, MiniSetup)]
-}
+import xsbti.compile.{ AnalysisContents, AnalysisStore }
 
 object AnalysisStore {
-  def cached(backing: AnalysisStore): AnalysisStore = new AnalysisStore {
-    private var last: Option[(CompileAnalysis, MiniSetup)] = None
-    def set(analysis: CompileAnalysis, setup: MiniSetup): Unit = {
-      backing.set(analysis, setup)
-      last = Some((analysis, setup))
+  def cached(backing: AnalysisStore): AnalysisStore = new CachedAnalysisStore(backing)
```

Comment:
I'm adding docs to `cached`, though. And also adding these options to the public API.